src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -143,18 +141,13 @@ export class StudyViewPageStore {
     }
 
     @action
-    updateGeneFilter(entrezGeneId: number) {
+    updateGeneFilters(entrezGeneIds: number[]) {
         let mutatedGeneFilter = this._mutatedGeneFilter;
         if (!mutatedGeneFilter) {
             //TODO: all elements instead of one
             mutatedGeneFilter = { entrezGeneIds: [] };
         }
-        let _index = mutatedGeneFilter.entrezGeneIds.indexOf(entrezGeneId);
-        if (_index === -1) {
-            mutatedGeneFilter.entrezGeneIds.push(entrezGeneId);
-        } else {
-            mutatedGeneFilter.entrezGeneIds.splice(_index, 1);
-        }
+        mutatedGeneFilter.entrezGeneIds = _.xor(mutatedGeneFilter.entrezGeneIds, entrezGeneIds);
```

Comment:
you might want to consider making a new gene filter object each time instead of treating the old one as mutable.  is there anything else on the filter object besides the list of geneIds?  if not, then maybe just the gene collection instead of nested object.

Reply:
this is just a general best practice because then you don't have to use deep observation on the collection.  there probably won't be a ton of genes here so it may be insignificant, but still

Reply:
then you could make the array observable ref